### DISCUSSION TITLE
Frontend Brainstorming

### DISCUSSION THREAD

**Discussion by Cruiz102** [uprm-inso4101-2023-2024-s2/semester-project-personalfinancemanager#3] [Ideas]
Hi everyone! I'm excited to be your manager for this project. The architecture of the project is still under development, but I would like to open this discussion page to gather ideas for designing the user interfaces and the frontend side of the application.

Core features we anticipate for the app:

- Adding incomes and expenses for a system that manages time (calendars, agendas, tables).
- Displaying financial behavior.
- Data visualization with graphs, tables, and statistics.
- User menu.

Please remember that some aspects are still undefined, and certain requirements might change in the future. However, these are the primary ideas we want to start with, especially in structuring the frontend part of the project.

topics for discussion (but not limited to):

- Software stack choices for web development.
- Design concepts using tools like Figma.
- Implementation strategies for the frontend.
- Potential development for mobile platforms (Android or iOS).

**Comment by fernandopizarro1** [comment]
We could separate into two different accounts, a personal and a business account. After creating a signup login page, they can be two different pages and have different options .  For example, the personal side can have an option such as “groceries” and the business side can have an option like “stocks” or “merchandise”.

**Reply by Goya01** [reply]
Why don't we use Figma to create a "prototype" or a design of what different features of the app would look like. For example: the main page, how would the visuals be accommodated, profiles, etc. We could then start implementing those features based on how we "designed" them to look like.

**Reply by SammyAlva04** [reply]
We could also include a display of any debt or things of the sort that a person has, maybe with reminders.

**Reply by MarkoSchuetz** [reply]
Going for Figma right away might be too early. Think about this: creating a design in Figma requires quite some work compared to e.g. writing down a usage scenario and using that to start the discussion. Of course, eventually you will want the Figma screen design(s), but you can accelerate the discussion by using a different starting point, one that's way easier to come by.

One goal that we have here is to maintain momentum. So, things to can do to move the discussion along that are little effort are preferred...
